include/boost/geometry/arithmetic/infinite_line_functions.hpp

```diff
@@ -76,19 +76,6 @@ side_value(model::infinite_line<Type> const& line, Point const& p)
     return side_value(line, geometry::get<0>(p), geometry::get<1>(p));
 }
 
-// Returns true for two lines which are supposed to be (close to) collinear
-// (which is not checked) and have a similar direction
-// (in practice up to 45 degrees, TO BE VERIFIED)
-// true: -----------------> p -----------------> q
-// false: -----------------> p <----------------- q
-template <typename Type>
-inline
-bool similar_direction(const model::infinite_line<Type>& p,
-                       const model::infinite_line<Type>& q)
-{
-    return p.a * q.a >= 0 && p.b * q.b >= 0;
-}
-
 template <typename Type>
```

Comment:
What could go wrong was that (fr example) `p.b` was `-1.00e-16` and `q.b` was `+1.00e-16` and therefore their product was negative, while the segments obviously had the same direction...